cibyl/plugins/openstack/sources/zuul/deployments/formatting.py

```diff
@@ -0,0 +1,58 @@
+"""
+#    Copyright 2022 Red Hat
+#
+#    Licensed under the Apache License, Version 2.0 (the "License"); you may
+#    not use this file except in compliance with the License. You may obtain
+#    a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#    License for the specific language governing permissions and limitations
+#    under the License.
+"""
+from string import Template
+
+from tripleo.insights.io import Topology
+
+
+class TopologyPrinter:
```

Comment:
I wonder if it wouldn't be better to have this functionality as a method of the Topology class, so that if we need to extend with more node types we have all the changes contained in one place.

Reply:
Yes, this can all also be implemented in the '\__str__' method of Topology. But I don't know, being this on the Cibyl side of things, I thought of giving it its own class here.

I agree that the other way is also possible, both will work on any case.